BWEnv/src/controller.cc

```diff
@@ -586,31 +587,36 @@ int Controller::getAttackFrames(int unitID) {
   return attackFrames;
 }
 
-void Controller::serializeFrameData(torchcraft::fbs::FrameDataT* data) {
-  std::ostringstream out;
-  if (prev_sent_frame == nullptr) {
-    out << *last_frame;
-    data->is_diff = false;
-  } else {
-    out << replayer::frame_diff(last_frame, prev_sent_frame);
-    data->is_diff = true;
+void Controller::serializeFrameData(torchcraft::fbs::FrameOrFrameDiffUnion& frameOrFrameDiff) {
+  frameOrFrameDiff.Reset();
+  {    
```

Comment:
Why the extra scope?

Reply:
Just indicating that the builder's relevant scope is limited. If that's more confusing than clarifying I'm happy to remove it.